src/App/Fossa/YamlDeps.hs

```diff
@@ -0,0 +1,196 @@
+toBuildData deps =
+  SourceUnitBuild
+    { buildArtifact = "default",
+      buildSucceeded = True,
+      buildImports = imports,
+      buildDependencies = buildDeps
+    }
+  where
+    (imports, buildDeps) = unzip . map (addEmptyDep . toImport) $ NE.toList deps
+
+    toImport :: ReferencedDependency -> Locator
+    toImport ReferencedDependency {..} =
+      Locator
+        { locatorFetcher = depTypeToFetcher locDepType,
+          locatorProject = locDepPackage,
+          locatorRevision = locDepVersion
+        }
+
+    addEmptyDep :: Locator -> (Locator, SourceUnitDependency)
+    addEmptyDep loc = (loc, SourceUnitDependency loc [])
+
+toAdditionalData :: NE.NonEmpty CustomDependency -> AdditionalDepData
+toAdditionalData deps = AdditionalDepData {userDefinedDeps = map tosrc $ NE.toList deps}
+  where
+    tosrc CustomDependency {..} =
+      SourceUserDefDep
+        { srcUserDepName = customPackage,
+          srcUserDepVersion = customVersion,
+          srcUserDepLicense = customLicense,
+          srcUserDepDescription = customDescription,
+          srcUserDepUrl = customUrl
+        }
+
+hasNoDeps :: YamlDependencies -> Bool
+hasNoDeps YamlDependencies{..} = all (0 ==)
+  [ length referencedDependencies,
+    length customDependencies
+  ]
+
+data YamlDependencies = YamlDependencies
+  { referencedDependencies :: [ReferencedDependency],
+    customDependencies :: [CustomDependency]
+  } deriving (Eq, Ord, Show)
+
+data ReferencedDependency = ReferencedDependency
+  { locDepPackage :: Text,
+    locDepType :: DepType,
+    locDepVersion :: Maybe Text
+  }
+  deriving (Eq, Ord, Show)
+
+data CustomDependency = CustomDependency
+  { customPackage :: Text,
+    customVersion :: Text,
+    customLicense :: Text,
+    customDescription :: Maybe Text,
+    customUrl :: Maybe Text
+  } deriving (Eq, Ord, Show)
+
+instance FromJSON YamlDependencies where
+  parseJSON = withObject "YamlDependencies" $ \obj ->
+    YamlDependencies <$ (obj .:? "version" >>= isMissingOr1)
+      <*> (obj .:? "referenced-dependencies" .!= [])
+      <*> (obj .:? "custom-dependencies" .!= [])
+    where
+      isMissingOr1 :: Maybe Int -> Parser ()
+      isMissingOr1 (Just x) | x /= 1 = fail $ "Invalid fossa-deps version: " <> show x
+      isMissingOr1 _ = pure ()
+
+depTypeParser :: Text -> Parser DepType
+depTypeParser text = case depTypeFromText text of
+  Just t -> pure t
+  Nothing -> fail $ "dep type: " <> unpack text <> " not supported"
+
+instance FromJSON ReferencedDependency where
+  parseJSON = withObject "ReferencedDependency" $ \obj ->
+    ReferencedDependency <$> obj .: "package"
+      <*> (obj .: "type" >>= depTypeParser)
+      <*> obj .:? "version"
+      <* forbidMembers "referenced dependencies" ["license", "description", "url"] obj
+
+instance FromJSON CustomDependency where
+  parseJSON = withObject "CustomDependency" $ \obj ->
+    CustomDependency <$> obj .: "package"
```

Comment:
I think we agreed upon changing `package` to `name` for all use dependency sections.